src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
+
+  static deleteBeast() {
+    Animal.alive = Animal.alive.filter((beast) => beast.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (beast.hidden !== true && beast instanceof Herbivore) {
+      beast.health -= 50;
+    }
+
+    if (beast.health < 50) {
```

Comment:
But if beast has 40 health - it still alive, isn't it? Wild nature is cruel, but let's give it a chance

Reply:
Check that animal has less or 0 health to remove it from the list